cellfinder/core/detect/filters/volume/ball_filter.py

```diff
@@ -316,17 +385,20 @@ def _walk(
                 tile_step_height,
                 inside_brain_tiles,
             ):
-                cube = volume[
-                    x : x + kernel.shape[0],
-                    y : y + kernel.shape[1],
-                    :,
-                ]
                 if _cube_overlaps(
-                    cube,
+                    volume,
+                    x,
+                    x + kernel.shape[0],
+                    y,
+                    y + kernel.shape[1],
                     overlap_threshold,
-                    THRESHOLD_VALUE,
+                    threshold_value,
                     kernel,
                 ):
-                    volume[ball_centre_x, ball_centre_y, middle_z] = (
-                        SOMA_CENTRE_VALUE
+                    volume[middle_z, ball_centre_x, ball_centre_y] = (
```

Comment:
Should we be worried about the `prange` here introducing non-deterministic in-place changes to `volume` when this is run in parallel?

Reply:
Good catch. The next commit fixes it by passing a copy of `volume` that is unchanged during `walk`. I didn't check for performance differences, with the `copy`, but I can't imagine it has a big impact.